packages/owlbot-bootstrapper/common-container/fetch-api-info.ts

```diff
@@ -0,0 +1,149 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Storage} from '@google-cloud/storage';
+import {ServiceConfigYaml, DriftApi, ReleaseLevel} from './interfaces';
+import yaml from 'js-yaml';
+import {
+  FileNotFoundError,
+  RepositoryFileCache,
+} from '@google-automations/git-file-utils';
+
+/**
+ * Function that gets api information from a public DRIFT bucket
+ * @param apiId the unique API ID
+ * @param storageClient an instance of Google Cloud Storage
+ * @returns DriftApi info
+ */
+async function getDriftMetadata(
+  apiId: string,
+  storageClient: Storage
+): Promise<DriftApi> {
+  const bucket = 'devrel-prod-settings';
+  const file = 'apis.json';
+  const contents = await storageClient.bucket(bucket)?.file(file)?.download();
+  let apis;
+  if (contents) {
+    apis = JSON.parse(contents.toString()).apis;
+  }
+
+  return extractApiInfoFromJson(apis, apiId);
+}
+/**
+ * Function that extracts information for a given API from apis.json
+ *
+ * @param apis the apis.json file downloaded from drift as a JSON object
+ * @param apiId the unique identifier of the API
+ * @returns DriftAPI the fields as described in Drift
+ */
+function extractApiInfoFromJson(apis: DriftApi[], apiId: string): DriftApi {
+  const shortName = apiId.split('.')[apiId.split('.').length - 2];
+  for (const api of apis) {
+    if (api.api_shortname?.includes(shortName)) {
```

Comment:
This seems like it'll match false positives if an api short name contains another one. e.g maybe a short name is `bigquery` and this will match `bigquery-storage`.

Reply:
Confirm why we wouldn't use exact matching.